### PR TITLE
Updated error returns on rmw_publish to comply with documentation

### PR DESCRIPTION
**Pull request by Lobotuerk**
Signed-off-by: lobotuerk <jtlorente@ekumenlabs.com>

### CODE REVIEW

**Review by Blast545 — APPROVED** [review]
LGTM with green CI

File: rmw_cyclonedds_cpp/src/rmw_node.cpp [hunk 1]
```diff
@@ -1505,9 +1505,9 @@ extern "C" rmw_ret_t rmw_publish(
   rmw_publisher_allocation_t * allocation)
 {
   static_cast<void>(allocation);    // unused
-  RET_NULL(publisher);
+  RMW_CHECK_ARGUMENT_FOR_NULL(publisher, RMW_RET_INVALID_ARGUMENT);
   RET_WRONG_IMPLID(publisher);
```
Comment (hidmic) [inline rmw_cyclonedds_cpp/src/rmw_node.cpp:1509]: @Lobotuerk consider changing this one to `RMW_CHECK_TYPE_IDENTIFIERS_MATCH` too.

Comment (Lobotuerk) [reply rmw_cyclonedds_cpp/src/rmw_node.cpp:1509]: Isn't ```RET_WRONG_IMPLID``` doing that?

Comment (Lobotuerk) [reply rmw_cyclonedds_cpp/src/rmw_node.cpp:1509]: Oh, but that macro is RMW side, not specific implementation one, makes sense, nvm

**Review by clalancette — APPROVED** [review]
Looks good with green CI.

**Review by hidmic — APPROVED** [review]
LGTM pending green CI

**Comment by Lobotuerk** [comment]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/12274/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/7248/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/9992/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/12165/)

**Pull request merged**